python/download_nhs.py

```diff
@@ -0,0 +1,140 @@
+    current_data["lastUpdatedAt"],
+    "%Y-%m-%dT%H:%M:%S.%fZ")
+
+to_extract = ['countries', 'regions', 'utlas', 'overview']
+
+
+def normalize(data, key):
+    if key in data:
+        return {i["date"]: i["value"] for i in data[key]}
+    return {}
+
+
+history_rows = []
+latest_rows = []
+for target in to_extract:
+    root = current_data[target]
+    for sector_key in root.keys():
+        sector_data = root[sector_key]
+        sector_name = sector_data["name"]["value"]
+
+        new_confirmed = normalize(sector_data, "dailyConfirmedCases")
+        total_confirmed = normalize(sector_data, "dailyTotalConfirmedCases")
+        new_deaths = normalize(sector_data, "dailyDeaths")
+        total_deaths = normalize(sector_data, "dailyTotalDeaths")
+
+        all_dates = (
+            set(new_confirmed.keys()) |
+            set(total_confirmed.keys()) |
+            set(new_deaths.keys()) |
+            set(total_deaths.keys())
+        )
+
+        history = [{
+            "date": date,
+            "sector_type": target,
+            "sector_key": sector_key,
+            "sector_name": sector_name,
+            "new_confirmed": new_confirmed.get(date, None),
+            "total_confirmed": total_confirmed.get(date, None),
+            "new_deaths": new_deaths.get(date, None),
+            "total_deaths": total_deaths.get(date, None)
+        } for date in all_dates]
+
+        history_rows += history
+
+        sector_total_cases = sector_data.get("totalCases", {}) \
+            .get("value", None)
+        sector_new_cases = sector_data.get("newCases", {}) \
+            .get("value", None)
+        sector_total_deaths = sector_data.get("deaths", {}) \
+            .get("value", None)
+        latest_rows += [{
+            "sector_type": target,
+            "sector_key": sector_key,
+            "sector_name": sector_name,
+            "cases": sector_total_cases,
+            "deaths": sector_total_deaths,
+            "new_cases": sector_new_cases
+        }]
+
+history_header = [
+    "date",
+    "sector_type",
+    "sector_key",
+    "sector_name",
+    "new_confirmed",
+    "total_confirmed",
+    "new_deaths",
+    "total_deaths"]
+with open('/tmp/nhs/history.csv', 'w', newline='') as csvfile:
+    spamwriter = csv.writer(
+        csvfile,
+        delimiter=',',
+        quotechar='"',
+        quoting=csv.QUOTE_MINIMAL)
+    spamwriter.writerow(history_header)
+    for row in history_rows:
+        spamwriter.writerow([row[x] for x in history_header])
+
+latest_header = [
+    "sector_type",
+    "sector_key",
+    "sector_name",
+    "cases",
+    "deaths",
+    "new_cases"
+]
+with open('/tmp/nhs/latest.csv', 'w', newline='') as csvfile:
+    spamwriter = csv.writer(
+        csvfile,
+        delimiter=',',
+        quotechar='"',
+        quoting=csv.QUOTE_MINIMAL)
+    spamwriter.writerow(latest_header)
+    for row in latest_rows:
+        spamwriter.writerow([row[x] for x in latest_header])
+
+for map in maps:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'map'](https://app.codacy.com/manual/jrtapsell/MetabaseAutoCovid/pullRequest?prid=5378392)